modules/imaging_uploader/js/imaging_uploader.js

```diff
@@ -155,3 +155,302 @@ $(function () {
+ */
+function UploadProgress() {
+    /**
+     * Row (i.e. <tr> element) in the MRI upload table associated to this progress element.
+     */ 
+    this. _uploadRow         = null;
+    /**
+     * Series of dots used in the text used to describe the progress object
+     * to indicate that the MRI pipeline is currently running.
+     */
+    this._dots               = '';
+    /**
+     * Char used in the text used to describe the progress object
+     * to indicate that the MRI pipeline is currently running.
+     */
+    this._animatedCharIndex  = 0;
+    /**
+     * Server response to a POST request for the status of an upload progress.
+     */
+    this._progressFromServer = null; 
+        
+    /**
+     * Getter method for field _uploadRow.
+     */    
+    this.getUploadRow = function() {
+        return this._uploadRow;
+    };
+    
+    /**
+     * Setter method for field _uploadRow.
+     */
+    this.setUploadRow = function(uploadRow) {
+        this._uploadRow = uploadRow;
+        
+        this._progressFromServer = null;
+        this._dots              = '';
+        this._animatedCharIndex = 0;
+    };
+
+    /**
+     * Gets the upload ID associated to the current progress object.
+     */
+    this.getUploadId = function() {
+        if(this._uploadRow == null) {
+            return null;
+        }
+        
+        return $.trim($(this._uploadRow).find('td').eq(1).text());
+    }    
+    
+    /**
+     * String representation of this object.
+     */
+    this.getProgressText = function() {
+        var columns    = $(this._uploadRow).find('td'),
+            uploadId   = $.trim($(columns).eq(1).text()),
+            pscid      = $.trim($(columns).eq(4).text()),
+            visitLabel = $.trim($(columns).eq(5).text())
+
+        var progressType = $('select[name=LogType] option:selected').text();
+
+        // All the messages in the notification_spool table
+        var notificationText = '';        
+        if(this._progressFromServer != null && this._progressFromServer.notifications != null) {
+            for(i=0; i<this._progressFromServer.notifications.length; i++) {
+                notificationText += this._progressFromServer.notifications[i].Message;
+            }
+        }
+                
+        // If pipeline is still running
+        if(this.isInserting()) {
+            return progressType + ' of upload ' + uploadId + ' for ' + pscid + ' at ' + visitLabel + "\n"
+            + notificationText
+            + this._dots
+            + ['|', '/', '-', '\\', '|', '/', '-', '\\'][this._animatedCharIndex % 8] + "\n";
+        }
+        
+        // Pipeline finshed running: failed or succeeded ?
+        var statusText = 'Upload is finished and ' + (this.isInsertionComplete() ? 'was successful' : 'failed');
+                
+        return progressType + ' of upload ' + uploadId + ' for ' + pscid + ' at ' + visitLabel + "\n"
+            + notificationText
```

Comment:
@nicolasbrossard,
it is a good idea to include the CandID on the message here (besides the PSCID and the Visit label)